refactor(skills): replace arch-diagram-builder agent with reusable architecture-diagrams skill (#2068)
microsoft/hve-core: .github/skills/project-planning/requirements-author/templates/brd/diagram-ascii.md

*Guidance*: Provide a brief narrative explanation of the process steps, decision points, and outcomes. Keep to 2-3 paragraphs.

## Optional: Technical Context via Architecture Diagrams Skill

{{arch_diagram_ref}}

> *Guidance*: When this process depends on infrastructure or system integration details, read the architecture-diagrams skill at `.github/skills/hve-core/architecture-diagrams/SKILL.md` with `read_file` and follow its workflow, conventions, arrow types, grouping, layout, resource identification, output format, worked example, and authoring guidelines. Otherwise, leave blank.

Reviewer: [portability] Hardcoded `read_file` path in BRD template guidance

The guidance text explicitly instructs `read the architecture-diagrams skill at .github/skills/hve-core/architecture-diagrams/SKILL.md with read_file`. This is the exact pattern the new `prompt-builder.instructions.md` rule prohibits: the `.github/skills/` root does not exist in plugin or extension distributions and a `read_file` on a repo-root path will fail there.

```suggestion
*Guidance*: When this process depends on infrastructure or system integration details, use the `architecture-diagrams` skill: load its `SKILL.md` and follow its authoring contract. Otherwise, leave blank.
```

Author: Done. The BRD template guidance now says to use the `architecture-diagrams` skill by name and follow its workflow, dropping the hardcoded `read_file` path that the `prompt-builder.instructions.md` rule prohibits. Pushed in 057e8313.
